app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt

```diff
@@ -86,6 +86,7 @@ class RegistrationActivity : AppCompatActivity() {
                             .setFullName(nameEditText.text.toString())
                             .setPassword(passwordEditText.text.toString())
                             .setUserName(nameEditText.text.toString())
+                            .setReferralCode(referralCodeEditText.text.toString())
```

Comment:
Shouldn't you be checking for `InvalidReferralCodeError` below-

`val message = when (response.statusCode) {`

And then depending on your flow, probably not take him back to login page.

(PS why am I even reviewing this xD)

Reply:
> Shouldn't you be checking for `InvalidReferralCodeError` below-
> 
> `val message = when (response.statusCode) {`
> 
> And then depending on your flow, probably not take him back to login page.
> 
> (PS why am I even reviewing this xD)

Oh yeah, the response code. Got it xD